gui/elements/qt/_layerPanel.py

```diff
@@ -71,7 +71,11 @@ def __init__(self, layers):
         self.setStyleSheet(styleSheet)
 
     def on_click(self):
-        self.layers.viewer.add_markers(empty((0, self.layers.viewer.max_dims)))
+        if self.layers.viewer.dimensions.max_dims == 0:
```

Comment:
When would this be the case?

Reply:
so there's an awkward corner case where we launch a window with an empty viewer and then want to add a new empty markers layer for annotation. First question is should this even be allowed? I can see an argument for saying no. If yes, then what should the dimensions and shape of that new layer be? Unsure how to handle - I was thinking of addressing in a future PR though

Reply:
Layers of one kind should not depend on another layer type. If one adds a markers layer, as first layer, then on has to define the dimensions, just as one would do for images.

Reply:
@royerloic so you're saying that it should be allowed, but only if `size=my_tuple` is also passed through? I can see that being handy.

Reply:
Sounds good I just created an issue #91 for us to implement this in a future PR